src/MembershipManager.sol

```diff
@@ -167,10 +176,12 @@ contract MembershipManager is Initializable, OwnableUpgradeable, PausableUpgrade
         (uint256 totalBalance, uint256 feeAmount) = _withdrawAndBurn(_tokenId);
 
         // transfer 'eEthShares' of eETH to the owner
+        membershipShares -= liquidityPool.sharesForAmount(totalBalance - feeAmount);
         eETH.transfer(msg.sender, totalBalance - feeAmount);
 
         if (feeAmount > 0) {
-            liquidityPool.withdraw(address(this), feeAmount);
+            uint256 feeShares = liquidityPool.withdraw(address(this), feeAmount);
```

Comment:
By this `withdraw`, the membership manager contract will receive ETH (not eETH), which is used to boost the MembershipNFT APR when rebasing happens. so it should not be counted as `membershipShares`

* ETH balance in MembershipManager https://etherscan.io/address/0x3d320286E014C3e1ce99Af6d6B00f0C1D63E3000